beacon-chain/rpc/eth/v1/validator/validator.go

```diff
@@ -168,7 +170,25 @@ func (vs *Server) ProduceAttestationData(ctx context.Context, req *v1.ProduceAtt
 
 // GetAggregateAttestation aggregates all attestations matching the given attestation data root and slot, returning the aggregated result.
 func (vs *Server) GetAggregateAttestation(ctx context.Context, req *v1.AggregateAttestationRequest) (*v1.AggregateAttestationResponse, error) {
-	return nil, errors.New("Unimplemented")
+	ctx, span := trace.StartSpan(ctx, "validatorv1.GetAggregateAttestation")
+	defer span.End()
+
+	allAtts := vs.AttestationsPool.AggregatedAttestations()
+	for _, att := range allAtts {
+		if att.Data.Slot == req.Slot {
+			root, err := att.Data.HashTreeRoot()
+			if err != nil {
+				return nil, status.Errorf(codes.Internal, "Could not get attestation data root: %v", err)
+			}
+			if bytes.Equal(root[:], req.AttestationDataRoot) {
+				return &v1.AggregateAttestationResponse{
```

Comment:
Returning the first match doesn't seem right.

Should the response be a list of attestation?

If not, shouldn't we return the most profitable one?

Reply:
It should return a single attestation. I agree we should likely return the one with the most aggregation bits though @rkapka